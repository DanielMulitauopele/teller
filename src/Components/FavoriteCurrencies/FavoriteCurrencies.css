.fave-currency {
  background: var(--MainGradient);
  height: 11rem;
  width: 100%;
  position: relative;
  text-align: center;
  color: var(--AlmostWhite);
}

.fave-name {
  font-size: 0.8rem;
  padding-top: 50px;
}

.fave-price {
  font-size: 2.2rem;
  letter-spacing: -1px;
  font-weight: 300;
}

.fave-price:first-letter {
  font-size: 1.2rem;
  padding-right: 5px;
}

.fave-price span {
  font-size: 1.2rem;
}

.price-progress {
  display: inline-block;
  font-size: 0.8rem;
}

.fave-currency .positive + .negative {
  height: 1rem;
  width: 0.5rem;
  transform: rotate(90deg);
  margin-bottom: -2px;
}

.fave-currency .negative {
  transform: rotate(-90deg);
}

.bubbles {
  width: 5rem;
  height: 2rem;
  bottom: 2rem;
  position: absolute;
  display: flex;
}

.bubbles-left {
  left: 4rem;
}

.bubbles-right {
  right: 3rem;
}

.bubbles div {
  margin: 8px;
}

.bubbles div:first-child,
.bubbles div:nth-child(3) {
  width: 5px;
  height: 5px;
  background-color: white;
  border-radius: 50%;
  margin-top: 25px;
}

.bubbles div:nth-child(2) {
  width: 10px;
  height: 10px;
  background-color: rgb(216, 216, 216);
  border-radius: 50%;
}

.bubbles-left div:first-child {
  animation-delay: 0.7s !important;
  bottom: 0;
  position: absolute;
}

.bubbles-left div:nth-child(2) {
  animation-delay: 1.1s !important;
  bottom: 0;
  left: 1rem;
  position: absolute;
}

.bubbles-left div:nth-child(3) {
  animation-delay: 1.4s !important;
  bottom: 15px;
  left: 3rem;
  position: absolute;
}

.bubbles-right div:first-child {
  animation-delay: 0.5s !important;
  bottom: 0;
  position: absolute;
}

.bubbles-right div:nth-child(2) {
  animation-delay: 0.9s !important;
  bottom: 0;
  left: 1rem;
  position: absolute;
}

.bubbles-right div:nth-child(3) {
  animation-delay: 1.2s !important;
  bottom: 15px;
  left: 3rem;
  position: absolute;
}

.up {
  animation: rise 4s linear infinite !important;
}

.speed-up-2 {
  animation: rise 3s linear infinite !important;
}

.speed-up-3 {
  animation: rise 1.5s linear infinite !important;
}

.speed-up-4 {
  animation: rise 0.5s linear infinite !important;
}

.down {
  animation: rise 4s reverse infinite !important;
}

.speed-down-2 {
  animation: rise 3s reverse infinite !important;
}

.speed-down-3 {
  animation: rise 1.5s reverse infinite !important;
}

.speed-down-4 {
  animation: rise 0.5s reverse infinite !important;
}

@keyframes rise {
  0% {
    bottom: 1rem;
    opacity: 0;
  }
  90% {
    opacity: 1;
  }
  100% {
    bottom: 5rem;
    opacity: 0;
  }
}
