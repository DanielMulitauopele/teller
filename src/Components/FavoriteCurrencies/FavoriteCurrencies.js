import React, { Component } from "react";
import "./FavoriteCurrencies.css";
import Green from "../../Assets/green.svg";

class FavoriteCurrencies extends Component {
  constructor({ props, favorite, removeFromFavorites, addToFavorites }) {
    super(props);
  }

  controlBubbleSpeed = percent_change => {
    let className = "";
    switch (true) {
      case percent_change >= 6:
        className = "speed-up-4";
        //direction up
        //all speeds divided by 4
        break;
      case 4 <= percent_change && percent_change < 6:
        className = "speed-up-3";
        //direction up
        //all speeds divided by 3
        break;
      case 2 <= percent_change && percent_change < 4:
        className = "speed-up-2";
        //direction up
        //all speeds divided by 2
        break;
      case 0 <= percent_change && percent_change < 2:
        className = "up";
        //everything stays how it is currently in the css file
        break;
      case -2 <= percent_change && percent_change < 0:
        className = "down";
        //direction down
        break;
      case -4 <= percent_change && percent_change < -2:
        className = "speed-down-2";
        //direction down
        //all speeds divided by 2
        break;
      case -6 <= percent_change && percent_change < -4:
        className = "speed-down-3";
        //direction down
        //all speeds divided by 3
        break;
      case percent_change <= -6:
        className = "speed-down-4";
        //direction down
        //all speeds divided by 4
        break;
      default:
        className = "";
    }
    return className;
  };

  render() {
    const { favorite } = this.props;
    return (
      <div className="fave-currency">
        <div className="bubbles bubbles-left">
          <div className={this.controlBubbleSpeed(favorite.percent_change)} />
          <div className={this.controlBubbleSpeed(favorite.percent_change)} />
          <div className={this.controlBubbleSpeed(favorite.percent_change)} />
        </div>
        <p className="fave-name">{favorite.name}</p>
        <p className="fave-price">${favorite.price}</p>
        <div className="bubbles bubbles-right">
          <div className={this.controlBubbleSpeed(favorite.percent_change)} />
          <div className={this.controlBubbleSpeed(favorite.percent_change)} />
          <div className={this.controlBubbleSpeed(favorite.percent_change)} />
        </div>
        <div className="price-progress">
          <p>
            {favorite.percent_change < 0 ? "-" : "+"}$
            {((favorite.percent_change < 0 ? -1 : 1) *
              Math.round(favorite.price * favorite.percent_change)) /
              100}{" "}
            ({favorite.percent_change}% <img className="arrow" src={Green} alt="" />)
          </p>
        </div>
      </div>
    );
  }
}

export default FavoriteCurrencies;
