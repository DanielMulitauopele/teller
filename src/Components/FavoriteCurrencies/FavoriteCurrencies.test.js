import React from 'react'
import FavoriteCurrencies from './FavoriteCurrencies'
import { shallow } from 'enzyme'

describe('FavoriteCurrencies Component', () => {
  let wrapper
  let mockFavorite = {
    name: 'Some Coin',
    price: 0.00000000000,
    percent_change: 1.0000000000
  }
  let mockRemoveFromFavorites = jest.fn()
  let mockAddToFavorites = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<FavoriteCurrencies
                        favorite={mockFavorite}
                        removeFromFavorites={mockRemoveFromFavorites}
                        addToFavorites={mockAddToFavorites}
                      />)
  })

  xit('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})