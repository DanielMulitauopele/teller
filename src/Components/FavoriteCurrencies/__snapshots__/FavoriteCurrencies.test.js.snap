// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCurrencies Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoriteCurrencies
    addToFavorites={[MockFunction]}
    favorite={
      Object {
        "name": "Some Coin",
        "percent_change": 1,
        "price": 0,
      }
    }
    removeFromFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="bubbles bubbles-left"
        >
          <div
            className="speed-up-3"
          />
          <div
            className="speed-up-3"
          />
          <div
            className="speed-up-3"
          />
        </div>,
        <p
          className="fave-name"
        >
          Some Coin
        </p>,
        <p
          className="fave-price"
        >
          $
          0
        </p>,
        <div
          className="bubbles bubbles-right"
        >
          <div
            className="speed-up-3"
          />
          <div
            className="speed-up-3"
          />
          <div
            className="speed-up-3"
          />
        </div>,
        <div
          className="price-progress"
        >
          <p>
            +
            $
            0
             
            (
            1
            % 
            <img
              className="arrow"
              src="green.svg"
            />
            )
          </p>
        </div>,
      ],
      "className": "fave-currency",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="speed-up-3"
            />,
            <div
              className="speed-up-3"
            />,
            <div
              className="speed-up-3"
            />,
          ],
          "className": "bubbles bubbles-left",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "speed-up-3",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "speed-up-3",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "speed-up-3",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Some Coin",
          "className": "fave-name",
        },
        "ref": null,
        "rendered": "Some Coin",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "$",
            0,
          ],
          "className": "fave-price",
        },
        "ref": null,
        "rendered": Array [
          "$",
          0,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="speed-up-3"
            />,
            <div
              className="speed-up-3"
            />,
            <div
              className="speed-up-3"
            />,
          ],
          "className": "bubbles bubbles-right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "speed-up-3",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "speed-up-3",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "speed-up-3",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            +
            $
            0
             
            (
            1
            % 
            <img
              className="arrow"
              src="green.svg"
            />
            )
          </p>,
          "className": "price-progress",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "+",
              "$",
              0,
              " ",
              "(",
              1,
              "% ",
              <img
                className="arrow"
                src="green.svg"
              />,
              ")",
            ],
          },
          "ref": null,
          "rendered": Array [
            "+",
            "$",
            0,
            " ",
            "(",
            1,
            "% ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "arrow",
                "src": "green.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            ")",
          ],
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="bubbles bubbles-left"
          >
            <div
              className="speed-up-3"
            />
            <div
              className="speed-up-3"
            />
            <div
              className="speed-up-3"
            />
          </div>,
          <p
            className="fave-name"
          >
            Some Coin
          </p>,
          <p
            className="fave-price"
          >
            $
            0
          </p>,
          <div
            className="bubbles bubbles-right"
          >
            <div
              className="speed-up-3"
            />
            <div
              className="speed-up-3"
            />
            <div
              className="speed-up-3"
            />
          </div>,
          <div
            className="price-progress"
          >
            <p>
              +
              $
              0
               
              (
              1
              % 
              <img
                className="arrow"
                src="green.svg"
              />
              )
            </p>
          </div>,
        ],
        "className": "fave-currency",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="speed-up-3"
              />,
              <div
                className="speed-up-3"
              />,
              <div
                className="speed-up-3"
              />,
            ],
            "className": "bubbles bubbles-left",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "speed-up-3",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "speed-up-3",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "speed-up-3",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Some Coin",
            "className": "fave-name",
          },
          "ref": null,
          "rendered": "Some Coin",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "$",
              0,
            ],
            "className": "fave-price",
          },
          "ref": null,
          "rendered": Array [
            "$",
            0,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="speed-up-3"
              />,
              <div
                className="speed-up-3"
              />,
              <div
                className="speed-up-3"
              />,
            ],
            "className": "bubbles bubbles-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "speed-up-3",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "speed-up-3",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "speed-up-3",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              +
              $
              0
               
              (
              1
              % 
              <img
                className="arrow"
                src="green.svg"
              />
              )
            </p>,
            "className": "price-progress",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "+",
                "$",
                0,
                " ",
                "(",
                1,
                "% ",
                <img
                  className="arrow"
                  src="green.svg"
                />,
                ")",
              ],
            },
            "ref": null,
            "rendered": Array [
              "+",
              "$",
              0,
              " ",
              "(",
              1,
              "% ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "arrow",
                  "src": "green.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              ")",
            ],
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
